src/dodal/beamlines/i23.py

```diff
@@ -93,3 +101,12 @@ def pilatus() -> PilatusDetector:
         drv_suffix="cam1:",
         fileio_suffix=HDF5_SUFFIX,
     )
+
+
+@device_factory()
+def detector() -> Positioner1D[I23DetectorPositions]:
```

Comment:
Should: I think calling it `detector` is a bit misleading as it doesn't have stuff to control the detector itself, just move it. In `i03.py` we call it `detector_motion`. When we have the rest of the detector in too we can think about combining them